split-the-stream/README.md

```diff
@@ -0,0 +1,89 @@
+# Ideas about splitting the source-git work and upstream work
+
+Some ideas, possibilities, pros and cons of moving out the source-git related work.
+
+## What does the source-git workflow mean?
+
+Must have:
+
+- If user creates a merge-request on the source-git repository:
+  - Create a matching merge-request to the dist-git repository.
+  - Sync the CI results from the dist-git merge-request to the source-git merge-request.
+- If the dist-git is updated, update the source-git repository.
```

Comment:
I would add: "... by opening a PR", or something similar. The main idea would be that the service should never directly push to a branch. All interactions with the repositories (both source-git and dist-git) should happen through PRs.

Reply:
Also for non-user dist-git updates? (Contributor's merge-request is merged to dist-git or mass rebuild or other external change.

Reply:
That's a good point. I would say, yes, even if in the end, those source-git MRs are merged automatically.

Reply:
Updated.